lib/core/src/Cardano/Wallet/Transaction.hs

```diff
@@ -69,6 +76,19 @@ data TransactionLayer t k = TransactionLayer
         -- This expects as a first argument a mean to compute or lookup private
         -- key corresponding to a particular address.
 
+    , mkCertificateTx
+        :: (Address -> Maybe (k 'AddressK XPrv, Passphrase "encryption"))
+        -> (PoolId, k 'AddressK XPrv)
```

Comment:
I wonder if we could extend the `Depth` with a `RewardAccountK` and work it out with some type families in the `HardDerivation` type class :thinking: ... (because `AddressK` and `RewardAccountK` are at the same depth-level so to speak, but their meaning is different because they have a different parent. 

Alternatively, make the key first parameter not a `Depth` but a `[Depth]` having keys in the form of:

```hs
k '[ 'UTxOExternal, 'AddressIx ] XPrv
k '[ 'RewardAccount, 'AccountIx ] XPrv
```

to clearly disambiguate them..